### ISSUE TITLE
Each block breaking inside inline block context

### ISSUE THREAD

**Issue by jake-a-wood** [zordius/lightncandy#241]
The PHP Code:
-------------

```php
require('./vendor/autoload.php');
use LightnCandy\LightnCandy;

// The Template:
$template = <<<VAREND
{{#>__layout__}}
	{{#*inline "content"}}
		This content has been overriden

		<ul>
			{{#each Data}}
			<li>{{.}}</li>
			{{/each}}
		</ul>

	{{/inline}}
{{/__layout__}}
VAREND;

// Partials:
$partials= array();
$partials['__layout__'] = <<<VAREND
<html>
<head>
	<title>LAYOUT</title>
</head>
<body>
	<div class="header">
		{{>nav}}
	</div>
	
	{{#>content}}
	Content is not overridden
	{{/content}}

	{{>footer}}
</body>
</html>
VAREND;
$partials['nav'] = <<<VAREND
<ul>
	<li>
		<a href="/">
			Home
		</a>
	</li>
	<li>
		<a href="/">
			About Us
		</a>
	</li>
	<li>
		<a href="/">
			What About You?
		</a>
	</li>
</ul>
VAREND;
$partials['footer'] = <<<VAREND
<div class="footer">
	<p>And then there was his feet</p>
</div>
VAREND;

$phpStr = LightnCandy::compile($template, array(
  // Used compile flags
  'flags' => LightnCandy::FLAG_RUNTIMEPARTIAL,
  'partials' => $partials,
));

echo "Generated PHP Code:\n$phpStr\n";

// Input Data:
$data = array(
  'Data' => [
		'Hey',
		'How', 
		'Are',
		'You'
	]
);

// Get the render function from the php file
$renderer = LightnCandy::prepare($phpStr);

echo "Result:\n" . $renderer($data);
```

The Issue:
----------

It looks like the each block within a block include and inline block causes the output of some slightly malformed php string.

After running the above you'll get an error to the likes of the following:

```
Parse error: parse error, expecting `'{'' in /private/var/folders/l7/9cdkk6s57sx3ycr56h7d4x0m0000gp/T/lci_sKTWyi on line 15
```
 
Using the each block outside this context produces the expected output. It's just in this context that it is breaking.

**Comment by jbyington** [comment]
Additional research seems to indicate the issue lies with line 191 of Partial.php.

`$code = preg_replace('/\bfunction\s*\((.*?)\)\s*{/', 'function(\\1)use($sp){', $code);`

Removal of the section

`use($sp)`

leads to the desired results of {{#each}}, but then produces new errors in rendering further up in the runtime.

`Notice: Undefined variable: sp in /private/var/folders/lv/kfdwbrd17bjf7b2ykplqqg3m0000gn/T/lci_HdxFCL on line 14
`

Despite this error, the template seemed to render correctly.

**Comment by jbyington** [comment]
To clarify, the error seems to be in the generated PHP code that is returned from the compile method.

in my test example, it referred to line 21, which reads:

`'.$sp.''.$sp.''.LR::debug('each [Data].[repeat].[members]', 'sec', $cx, ((isset($in['Data']['repeat']) && is_array($in['Data']['repeat']) && isset($in['Data']['repeat']['members'])) ? $in['Data']['repeat']['members'] : LR::miss($cx, '[Data].[repeat].[members]')), null, $in, true, function($cx, $in)use($sp)use($sp){return '    Hello '.LR::debug('this', 'encq', $cx, $in).'
`

Notice the following part:

`function($cx, $in)**use($sp)use($sp)**{return `

That seems to be the issue, which lead me to alter Partial.php at line 191 to get the desired result.

Thank you.

**Comment by jake-a-wood** [comment]
After looking into jbyington's insights I think the solution might be as simple as changing the following in Partial.php on like 192 from: 

```
$code = preg_replace('/\bfunction\s*\((.*?)\)\s*{/', 'function(\\1)use($sp){', $code);
```

to 

```
$code = preg_replace('/\bfunction\s*\(([^\)]*)\)[^use]*\{/', 'function(\\1)use($sp){', $code);
```

The idea being to improve the regex to not match on functions that already have the use statements in them.

**Comment by jbyington** [comment]
I made this change and it corrected the problem. Thanks

**Comment by zordius** [comment]
This should be fixed in current master branch, for composer please try `dev-master` or `1.0.0-dev` , thanks.

**Comment by zordius** [comment]
the fixed is released in 1.0.0 now
